lib/src/query.c

```diff
@@ -3737,7 +3746,11 @@ static inline bool ts_query_cursor__advance(
           // pattern.
           bool node_does_match = false;
           if (step->symbol == WILDCARD_SYMBOL) {
-            node_does_match = !node_is_error && (is_named || !step->is_named);
+            if (step->is_missing) {
+              node_does_match = is_missing;
+            } else {
+              node_does_match = !node_is_error && (is_named || !step->is_named);
+            }
           } else {
             node_does_match = symbol == step->symbol;
           }
```

Comment:
One last question, didn't we want to support specific missing nodes too? like `(MISSING identifier)`?

Reply:
Of course :) Yeah that would be really helpful... would you prefer I work on it on this PR or try in a second one?

Reply:
Ok I have just gone ahead and implemented this (in a third commit, for easier change viewing)